cobalt/user_agent_platform_info.cc

```diff
@@ -0,0 +1,602 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/user_agent_platform_info.h"
+
+#include <map>
+#include <memory>
+#include "base/command_line.h"
+#include "base/logging.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_util.h"
+#include "base/strings/stringprintf.h"
+#include "starboard/common/system_property.h"
+#include "starboard/extension/platform_info.h"
+#include "starboard/system.h"
+#if SB_IS(EVERGREEN)
+#include "chrome/updater/util.h"
+#include "starboard/extension/installation_manager.h"
+#endif  // SB_IS(EVERGREEN)
+
+namespace cobalt {
+
+void GetUserAgentInputMap(
+    const std::string& user_agent_input,
+    std::map<std::string, std::string>& user_agent_input_map) {
+  struct state {
```

Comment:
This should be a class [1]. If the code is just forward-ported  
leave a TODO(bug) : )


[1] https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes

Reply:
I think this is fine to be a struct as a very simple data structure.